editors/rwa/portfolio.tsx

```diff
@@ -1,67 +1,126 @@
-const assetTypeOptions = [
-    { id: '91279GF8', label: 'T-Bill 91279GF8' },
-    { id: '91279GF9', label: 'T-Bill 91279GF9' },
-];
+const fieldsPriority: (keyof FixedIncomeAsset)[] = [
+    'id',
+    'name',
+    'maturity',
+    'notional',
+    'coupon',
+    'purchasePrice',
+    'purchaseDate',
+    'totalDiscount',
+    'purchaseProceeds',
+] as const;
 
-// TODO: replace with real data
-const maturityOptions = [
-    { id: 'purchase', label: 'Purchase' },
-    { id: 'mature', label: 'Mature' },
-];
+export const columnCountByTableWidth = {
+    1520: 12,
+    1394: 11,
+    1239: 10,
+    1112: 9,
+    984: 8,
+} as const;
+
+function createAssetFromFormInputs(data: RWAAssetDetailInputs) {
+    const maturity = data.maturity.toString() + 'T00:00:00.000Z';
+    return {
+        ...data,
+        maturity,
+    };
+}
 
-export const Portfolio = () => {
-    const [expandedRowId, setExpandedRowId] = useState<string | null>(null);
-    const [editRowId, setEditRowId] = useState<string | null>(null);
+export const Portfolio = (props: IProps) => {
+    const [expandedRowId, setExpandedRowId] = useState<string>();
+    const [selectedAssetToEdit, setSelectedAssetToEdit] =
+        useState<FixedIncomeAsset>();
     const [showNewAssetForm, setShowNewAssetForm] = useState(false);
 
-    const onClickDetails: FixedIncomeAssetsTableProps['onClickDetails'] =
-        item => {
-            setExpandedRowId(
-                item.id === expandedRowId ? null : item.id || null,
-            );
-        };
+    const { dispatch, document } = props;
 
-    const onEditItem: FixedIncomeAssetsTableProps['onEditItem'] = item =>
-        setEditRowId(item.id || null);
+    const spvs = document.state.global.spvs;
 
-    const onCancelEdit: FixedIncomeAssetsTableProps['onCancelEdit'] = reset => {
-        reset();
-        setEditRowId(null);
-    };
+    const fixedIncomeTypes = document.state.global.fixedIncomeTypes;
 
-    const onSubmitEdit: FixedIncomeAssetsTableProps['onSubmitEdit'] = (
-        data,
-        reset,
-    ) => {
-        // TODO: dispatch edit asset action
-        console.log('onSubmitEdit', data);
-        console.log('reset', reset);
-        setEditRowId(null);
-    };
+    const portfolio = document.state.global.portfolio
+        .filter((asset): asset is FixedIncome => isFixedIncomeAsset(asset))
+        .map(item => ({
+            ...item,
+            maturity: item.maturity.split('T')[0],
+            purchaseDate: item.purchaseDate.split('T')[0],
+        })) as FixedIncomeAsset[];
 
-    const onSubmitCreate: FixedIncomeAssetsTableProps['onSubmitEdit'] = (
-        data,
-        reset,
-    ) => {
-        // TODO: dispatch create asset action
-        console.log('onSubmitCreate', data);
-        console.log('reset', reset);
-        setShowNewAssetForm(false);
-    };
+    const toggleExpandedRow = useCallback(
+        (id: string) => {
+            setExpandedRowId(id === expandedRowId ? undefined : id);
```

Comment:
Small nitpick but here I would use an updater function instead.
`setExpanderRowId(expandedRowId => id === expandedRowId ? undefined : id);`
Then the dependency on `expandedRowId` can be removed from the hook.

Reply:
all good, will update